public/app/core/components/SplitPaneWrapper/SplitView.tsx

```diff
@@ -0,0 +1,93 @@
+import { css } from '@emotion/css';
+import { useViewportSize } from '@react-aria/utils';
+import React, { ReactNode } from 'react';
+import SplitPane from 'react-split-pane';
+
+import { GrafanaTheme2 } from '@grafana/data';
+import { useStyles2 } from '@grafana/ui';
+
+interface Props {
+  children: [ReactNode, ReactNode];
+  uiState: { rightPaneSize: number };
+  minSize?: number;
+  onResize?: (size: number) => void;
+}
+
+const onDragFinished = (size: number, onResize?: (size: number) => void) => {
+  document.body.style.cursor = 'auto';
+  onResize?.(size);
+};
+
+const onDragStarted = () => {
+  document.body.style.cursor = 'row-resize';
+};
+
+const getResizerStyles = (hasSplit: boolean) => (theme: GrafanaTheme2) =>
+  css`
+    position: relative;
+    display: ${hasSplit ? 'block' : 'none'};
+
+    &::before {
+      content: '';
+      position: absolute;
+      transition: 0.2s border-color ease-in-out;
+      border-right: 1px solid ${theme.colors.border.weak};
+      height: 100%;
+      left: 50%;
+      transform: translateX(-50%);
+    }
+
+    &::after {
+      background: ${theme.colors.border.weak};
+      content: '';
+      position: absolute;
+      left: 50%;
+      top: 50%;
+      transition: 0.2s background ease-in-out;
+      transform: translate(-50%, -50%);
+      border-radius: 4px;
+      height: 200px;
+      width: 4px;
+    }
+
+    &:hover {
+      &::before {
+        border-color: ${theme.colors.primary.main};
+      }
+
+      &::after {
+        background: ${theme.colors.primary.main};
+      }
+    }
+
+    cursor: col-resize;
+    width: ${theme.spacing(2)};
+  `;
+
+export const SplitView = ({ uiState: { rightPaneSize }, children, minSize = 200, onResize }: Props) => {
+  const { width } = useViewportSize();
+
+  // create two elements for library, even if only one exists (one will be hidden)
+  const hasSplit = children.filter(Boolean).length === 2;
+
+  const existingChildren = [
```

Comment:
@Elfo404 This is how I propose getting around the library limitation. This is an addition to the edits you provided. This allows us to key the panes so React doesn't complain :)

Reply:
@gelicia this is a little bit bad refactor as the resizer styles are now duplicated in SplitPaneWrapper and SplitView (and causing some bugs that where hard to track down, the horizontal reizer has a border now)

Reply:
@torkelo We did duplicate the styles yes, but the reasoning was that the SplitPaneWrapper had a lot of logic in it that was specific to dashboards, and we wanted another layer of abstraction that would keep things like styles that would be used universally in Grafana but without the horizontal and vertical splits that are used in dashboards. I realize now I should have brought the dashboards team in to review this directly. My apologies.

Reply:
@gelicia no worries

Reply:
> wanted another layer of abstraction

But the horizontal split was still left in SplitPaneWrapper, so not really another level of abstraction, just a new duplicated abstraction for vertical splits, one that now has a different (and duplicated) design and behaviors (And missing features for persistance) 

https://github.com/grafana/grafana/pull/56190